app/gui2/src/components/GraphEditor/GraphNode.vue

```diff
@@ -179,24 +179,23 @@ const isOutputContextOverridden = computed({
   set(shouldOverride) {
     const module = projectStore.module
     if (!module) return
-    const replacements = shouldOverride
-      ? [Ast.TextLiteral.new(projectStore.executionMode)]
-      : undefined
     const edit = props.node.rootSpan.module.edit()
-    const newAst = prefixes.modify(
-      edit,
-      props.node.rootSpan,
-      projectStore.isOutputContextEnabled
-        ? {
-            enableOutputContext: undefined,
-            disableOutputContext: replacements,
-          }
-        : {
-            enableOutputContext: replacements,
-            disableOutputContext: undefined,
-          },
-    )
-    graph.setNodeContent(props.node.rootSpan.exprId, newAst.code())
+    const replacementText = shouldOverride
+      ? [Ast.TextLiteral.new(projectStore.executionMode, edit)]
+      : undefined
+    const replacements = projectStore.isOutputContextEnabled
+      ? {
+          enableOutputContext: undefined,
+          disableOutputContext: replacementText,
+        }
+      : {
+          enableOutputContext: replacementText,
+          disableOutputContext: undefined,
+        }
+    const expression = props.node.rootSpan
+    const newAst = prefixes.modify(edit, expression, replacements)
+    const code = newAst.code()
+    graph.setNodeContent(props.node.rootSpan.exprId, code)
```

Comment:
Inertia (this was how it was originally implemented). Also, we don't have an `Owned<Ast>` here, which is required by `setExpression`.

Reply:
Naively taking the `AST` does not seem to work. The following code causes an infinite recursion somewhere in the AST.
```
    const ownedAst = edit.take(newAst.exprId)
    if (ownedAst === undefined) throw new Error('Failed to take new expression')
    graph.setExpression(props.node.rootSpan.exprId, ownedAst.node)
```

Reply:
Oh that's true. To reuse the template AST, we'll need an operation like `MutableModule.copy(ast: Ast): Owned<Ast>` that obtains ownership by assigning all new IDs.

Reply:
Does it have to be a `MutableModule`? I don't know the AST system very well yet, but since we are not modifying the module, should this not work with a `Module`? Or is the act of creating IDs a module manipulation?

Reply:
It would be a method on the module that we want to copy the subtree into.